x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/io/stream/PlanStreamOutputTests.java

```diff
@@ -200,39 +202,23 @@ public void testWriteMultipleAttributesWithSmallCache() throws IOException {
             BytesStreamOutput out = new BytesStreamOutput();
             PlanStreamOutput planStream = new PlanStreamOutput(out, PlanNameRegistry.INSTANCE, configuration, PlanNamedTypes::name, 10)
         ) {
-            List<Attribute> attrs = new ArrayList<>();
-            int occurrences = randomIntBetween(2, 300);
-            for (int i = 0; i < occurrences; i++) {
-
-                attrs.add(randomAttribute());
-            }
-
-            // send all the attributes, three times
-            for (int i = 0; i < 3; i++) {
-                for (Attribute attr : attrs) {
-                    planStream.writeNamedWriteable(attr);
-                }
-            }
-
-            try (PlanStreamInput in = new PlanStreamInput(out.bytes().streamInput(), PlanNameRegistry.INSTANCE, REGISTRY, configuration)) {
-                List<Attribute> readAttrs = new ArrayList<>();
-                for (int i = 0; i < occurrences; i++) {
-                    readAttrs.add(in.readNamedWriteable(Attribute.class));
-                    assertThat(readAttrs.get(i), equalTo(attrs.get(i)));
+            expectThrows(IllegalStateException.class, () -> {
```

Comment:
Thought: ISE here is reasonable; but there could just be rude queries that trigger us to go above the limit, which should become client errors (400) instead of ISE (500); what do you think?

Reply:
Good point, let's avoid 500s.
Probably `InvalidArgumentException` is appropriate